pr-deploy/test/app.test.ts

```diff
@@ -17,115 +17,145 @@
+      .post('/repos/test-owner/test-repo/check-runs', body => {
+        expect(body).toMatchObject({
+          'head_sha': 'abcde',
+          'name': 'test-check',
+          'status': 'queued',
+        });
+        return true;
+      })
+      .reply(200);
+
+    await probot.receive(prOpenedEvent);
   });
 
-  test('refreshes the check when a pull request is synchronized or reopened',
-    async() => {
-      // make sure a check already exists
-      github.checks.listForRef.mockReturnValue(
-        {data: {total_count: 1, check_runs: [{id: 12345}]}}
-      );
-
-      const prSynchronizedEvent:
-      Webhooks.WebhookEvent<WebhookPayloadPullRequest> = {
-        id: 'prId',
-        name: 'pull_request.synchronize',
-        payload: {
-          pull_request: {head: {sha: 'abcde'}},
-          repository: {owner: {name: 'repoOwner'},name: 'existingRepo'},
-        } as WebhookPayloadPullRequest,
-      };
-
-      await app.receive(prSynchronizedEvent);
-      expect(github.checks.update).toHaveBeenCalled();
-      expect(github.checks.create).not.toHaveBeenCalled();
-    });
-
-  test('enables deployment action when post is received', async done => {
-    // make sure a check already exists
-    github.checks.listForRef.mockReturnValue(
-      {data: {total_count: 1, check_runs: [{id: 12345}]}}
-    );
+  test('refreshes the check when a pull request is ' +
+  'synchronized or reopened', async() => {
+    nock(apiUrl)
+      .get('/repos/test-owner/test-repo/commits/' +
+      'abcde/check-runs?check_name=test-check')
+      .reply(200, {data: {total_count: 1, check_runs: [{id: 12345}]},
+      }) // make sure a check already exists
+      .post('/repos/test-owner/test-repo/check-runs', body => {
+        expect(body).toMatchObject({
+          'head_sha': 'abcde',
+          'name': 'test-check',
+          'status': 'queued',
+        });
+        return true;
+      })
+      .reply(200);
+
+    const prSynchronizedEvent: Webhooks.WebhookEvent<
+    Webhooks.EventPayloads.WebhookPayloadPullRequest> = {
+      id: 'prId',
+      name: 'pull_request.synchronize',
+      payload: {
+        pull_request: {head: {sha: 'abcde'}},
+        repository: {owner: {name: 'test-owner'}, name: 'test-repo'},
+      } as Webhooks.EventPayloads.WebhookPayloadPullRequest,
+    };
 
-    request(server)
-      .post('/v0/pr-deploy/travisbuilds/1/headshas/3/success')
-      .expect(() => { expect(github.checks.update).toHaveBeenCalledTimes(1);})
-      .expect(200, done);
+    await probot.receive(prSynchronizedEvent);
   });
 
   test('deploys the PR check when action is triggered', async() => {
-    // make sure a check already exists
-    github.checks.listForRef.mockReturnValue(
-      {data: {total_count: 1, check_runs: [
-        {id: 12345, output: {text: 'Travis build number: 3'}},
-      ]}}
-    );
-
-    const requestedActionEvent:
-    Webhooks.WebhookEvent<WebhookPayloadCheckRun> = {
+    nock(apiUrl)
+      .persist()
```

Comment:
I noticed we `persist()` here but not elsewhere. Is that because this endpoint is called more than once?

Reply:
Yep, but I changed it to use an exact number of times